tools/run_tests/run_xds_tests.py

```diff
@@ -1856,17 +1891,13 @@ def __init__(self, compute, alpha_compute, project):
                 result.state = 'FAILED'
                 result.message = str(e)
             finally:
-                if client_process and not client_process.returncode:
+                if client_process and client_process.returncode is None:
                     client_process.terminate()
-                test_log_file.close()
                 # Workaround for Python 3, as report_utils will invoke decode() on
                 # result.message, which has a default value of ''.
                 result.message = result.message.encode('UTF-8')
                 test_results[test_case] = [result]
-                if args.log_client_output:
-                    logger.info('Client output:')
-                    with open(test_log_filename, 'r') as client_output:
-                        logger.info(client_output.read())
+                client_logged.wait()
```

Comment:
nit: Do we want a deadline here? What should we do if client hang?

Reply:
Good point. We now wait a bit, send SIGKILL, wait again, and then just continue on if it fails to log.